eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Plutus/Evaluate.hs

```diff
@@ -91,6 +93,38 @@ instance (EraTxCert era, DecCBOR (ContextError era)) => DecCBOR (CollectError er
       dec 3 = SumD BadTranslation <! From
       dec n = Invalid n
 
+instance
+  ( Era era
+  , ToJSON (ScriptPurpose era)
+  , ToJSON (ContextError era)
+  ) =>
+  ToJSON (CollectError era)
+  where
+  toJSON = \case
+    NoRedeemer sPurpose ->
+      object
+        [ "kind" .= String "CollectError"
+        , "error" .= String "NoRedeemer"
+        , "plutusPurpose" .= toJSON sPurpose
+        ]
+    NoWitness sHash ->
+      object
+        [ "kind" .= String "CollectError"
+        , "error" .= String "NoWitness"
+        , "scriptHash" .= toJSON sHash
+        ]
+    NoCostModel lang ->
+      object
+        [ "kind" .= String "CollectError"
+        , "error" .= String "NoCostModel"
+        , "language" .= toJSON lang
+        ]
+    BadTranslation err ->
+      object
+        [ "kind" .= String "BadTranslation"
+        , "error" .= toJSON err
+        ]
```

Comment:
We could use `kindObject` here, though probably it's the same number of characters in the end.